src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3643,35 +3652,31 @@ bool Creature::IsNotReachableAndNeedRegen() const
     return false;
 }
 
-time_t Creature::GetLastDamagedTime() const
+std::shared_ptr<time_t> const& Creature::GetLastLeashExtensionTimePtr() const
 {
-    if (!_lastDamagedTime)
-        return time_t(0);
+    if (m_lastLeashExtensionTime == nullptr)
+        m_lastLeashExtensionTime = std::make_shared<time_t>(time(nullptr));
```

Comment:
Use `GameTime::GetGameTime().count()` instead of `time(nullptr)` (with time(nullptr) we are probably making a system call but GameTime::GetGameTime() has cached value - so no syscall).

Reply:
Added in https://github.com/azerothcore/azerothcore-wotlk/pull/19923.